### PR TITLE
py-aiodns: new port, version 1.1.1

### PR DESCRIPTION
**Pull request by IPGlider**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.6 17G65
Xcode 9.4.1 9F2000 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-aiodns/Portfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-aiodns
+version             1.1.1
+categories-append   devel
+platforms           darwin
+license             MIT
+
+python.versions     27 34 35 36 37
```
Comment (mf2k) [inline python/py-aiodns/Portfile]: Unless there is a specific need for them, remove the older python versions py34 and py35.

**Comment by pmetzger** [comment]
@IPGlider if you can fix @yan12125's issues, we can merge.

**Comment by reneeotten** [comment]
regarding dependencies: it also seems to require ```py-typing```, see https://github.com/saghul/aiodns/blob/master/setup.py#L24

**Comment by yan12125** [comment]
> regarding dependencies: it also seems to require py-typing

Good catch! I can't believe I missed that, which is just a few lines above the line I looked at :D

Technically, typing is only necessary for Python < 3.5 as it's a builtin module after that. For this port, you can add the dependency unconditionally or patch `setup.py` to put `typing` to extras_require and put the dependency in a `if` block for this port.

**Comment by reneeotten** [comment]
@yan12125 true, you don't need it for later Python versions and that will actually cause a problem here. Adding the dependency here unconditionally will not work as ```py-typing``` does not exist for the py36/py37 subports, but it should for sure be added in the case of py27. It seems to build regardless (as Travis doesn't complain about it), so I think only adding it to py27 and not patching ```setup.py``` should be fine here, or?

**Comment by yan12125** [comment]
If `setup.py` is not patched, the only issue I know is that `pip check` will report a missing dependency. Not a big issue, though, so it's fine skip that for this PR.

**Comment by pmetzger** [comment]
Merged. Thank you, @IPGlider!

**Pull request merged**